### DISCUSSION TITLE
fastify.close() takes over a minute if one GET request is made

### DISCUSSION THREAD

**Discussion by AmosSzust** [fastify/fastify#5010] [Q&A] [answered]
I created the simplest server using fastify with one get('/') route. On SIGTERM/SIGINT I do `fastify.close()`. To close the server I press the red rectangle (webstorm).

If no requests were made, closing is done immediately. If I do one simple GET request that is completed successfully, it takes over a minute until `fastify.close()` reaches its callback and even then, it looks like a timeout/force and not because of a graceful shutdown(?).

I tried keepAliveTimeout but it didn't change anything.

```
const fastify = Fastify({
  logger: true,
  keepAliveTimeout: 10000,
});

```
Am I missing something?

**Comment by jsumners** [comment]
+ https://fastify.dev/docs/latest/Reference/Server#keepalivetimeout
+ https://fastify.dev/docs/latest/Reference/Server#forcecloseconnections

**Reply by AmosSzust** [reply]
> The number of milliseconds of inactivity a server needs to wait for additional incoming data, after it has finished writing the last response, before a socket will be destroyed. If the server receives new data before the keep-alive timeout has fired, it will reset the regular inactivity timeout, i.e., [server.timeout](https://nodejs.org/api/http.html#servertimeout).

I thought this was the problem so I set `keepAliveTimeout: 10000` but it didn't change anything.

> When set to true, upon [close](https://fastify.dev/docs/latest/Reference/Server/#close) the server will iterate the current persistent connections and [destroy their sockets](https://nodejs.org/dist/latest-v16.x/docs/api/net.html#socketdestroyerror).

I don't want to foce closing connections, I want it to wait for them to close and then close the server itself. Not sure why and which connection remains open that prevents the server from closing.

**Reply by jsumners** [reply]
Your client is establishing a keep alive connection. You have to terminate the client's connection on either end.

**Reply by AmosSzust** [reply]
It's a browser's (Edge) url, a simple `localhost:8090` , does the browser establish a keep alive connection?

**Reply by jsumners** [reply, marked as answer]
Yes.

**Reply by AmosSzust** [reply]
grrr... I now tried with a rest client extension and indeed it was closed immediately even after requesting. Thanks,
